src/dependency/fileDownloader.ts

```diff
@@ -0,0 +1,84 @@
+import * as crypto from "crypto";
+import * as https from 'https';
+import { docsChannel } from '../common/shared';
+import { parse as parseUrl } from 'url';
+
+export async function downloadFile(description: string, urlString: string, integrity: string): Promise<Buffer> {
+    docsChannel.append(`Downloading package '${description}' `);
+
+    const url = parseUrl(urlString);
+    // TODO: Apply network settings(proxy, strictSSL..)
+    const options: https.RequestOptions = {
+        host: url.hostname,
+        path: url.path,
+        port: url.port
+    };
+
+    let buffers: any[] = [];
+
+    return new Promise<Buffer>((resolve, reject) => {
+        let request = https.request(options, response => {
+            if (response.statusCode === 301 || response.statusCode === 302) {
+                // Redirect - download from new location
+                return resolve(downloadFile(description, response.headers.location!, integrity));
+            }
+            else if (response.statusCode !== 200) {
+                // Download failed
+                return reject(new Error(`Failed to download from ${urlString}. Error code '${response.statusCode}'`));
+            }
+
+            let packageSize = parseInt(response.headers['content-length']!, 10);
+            let downloadedBytes = 0;
+            let downloadPercentage = 0;
+
+            docsChannel.append(`(${Math.ceil(packageSize / 1024)} KB) `);
+
+            response.on('data', data => {
+                downloadedBytes += data.length;
+                buffers.push(data);
```

Comment:
this is an in-efficient way of downloading large binary files in node, consider pipe it directly to a file:

Reply:
There are the reasons why I use Buffer instead of save to a real file:
 - we know the files size we are going to download, it is not a large file(<50M)
 - It is a efficient way, we can save `FileWrite` (when downloading), `FilerRead` (when unzipping) and `FileDelete`(when integrity check failed) operations

---
In reply to: [360265209](https://github.com/docascode/vscode-docs-build/pull/4#discussion_r360265209) [](ancestors = 360265209)

Reply:
Perf test:
![image](https://user-images.githubusercontent.com/17958546/71456363-09409980-27d4-11ea-96a9-679b2dcd9e51.png)